### PR TITLE
Build Astropy 3.2.3 for Pyodide

### PR DESCRIPTION
**Pull request by joseph-long**
The good news is, the build succeeds!

The bad news is, it doesn't seem to chain importing NumPy:

```
>>> import astropy.units
Error: Traceback (most recent call last):
  File "/lib/python3.7/site-packages/pyodide.py", line 42, in eval_code
    exec(compile(mod, '<exec>', mode='exec'), ns, ns)
  File "<exec>", line 1, in <module>
  File "/lib/python3.7/site-packages/astropy/__init__.py", line 121, in <module>
    _check_numpy()
  File "/lib/python3.7/site-packages/astropy/__init__.py", line 115, in _check_numpy
    raise ImportError(msg)
ImportError: Numpy version 1.13.0 or later must be installed to use Astropy
```

And depending on how you import things, they could work:

```
>>> import numpy 
>>> from astropy import units as u 
>>> u.m
Error: Traceback (most recent call last):
 File "/lib/python3.7/site-packages/pyodide.py", line 44, in eval_code 
     return eval(compile(expr, '<eval>', mode='eval'), ns, ns)
  File "<eval>", line 1, in <module>
NameError: name 'u' is not defined
>>> import astropy.units
>>> astropy.units.m
Unit("m")
```

Or *really* not:

```
Welcome to the Pyodide terminal emulator 🐍
>>> import astropy
Error: Out of bounds memory access (evaluating 'Module["asm"]["__runPython"].apply(null,arguments)')
>>> import astropy
[USER]: Out of bounds memory access (evaluating 'Module["asm"]["__pyproxy_apply"].apply(null,arguments)')
<?>.wasm-function[12245]@[wasm code]
<?>.wasm-function[12248]@[wasm code]
<?>.wasm-function[1748]@[wasm code]
<?>.wasm-function[15459]@[wasm code]
<?>.wasm-function[456]@[wasm code]
<?>.wasm-function[463]@[wasm code]
<?>.wasm-function[492]@[wasm code]
<?>.wasm-function[491]@[wasm code]
<?>.wasm-function[3224]@[wasm code]
<?>.wasm-function[3225]@[wasm code]
<?>.wasm-function[2681]@[wasm code]
<?>.wasm-function[15871]@[wasm code]
<?>.wasm-function[839]@[wasm code]
<?>.wasm-function[840]@[wasm code]
<?>.wasm-function[2749]@[wasm code]
<?>.wasm-function[2744]@[wasm code]
<?>.wasm-function[20461]@[wasm code]
<?>.wasm-function[835]@[wasm code]
<?>.wasm-function[831]@[wasm code]
<?>.wasm-function[830]@[wasm code]
<?>.wasm-function[847]@[wasm code]
<?>.wasm-function[1977]@[wasm code]
<?>.wasm-function[15600]@[wasm code]
<?>.wasm-function[837]@[wasm code]
<?>.wasm-function[2749]@[wasm code]
<?>.wasm-function[2744]@[wasm code]
<?>.wasm-function[20461]@[wasm code]
<?>.wasm-function[835]@[wasm code]
<?>.wasm-function[838]@[wasm code]
<?>.wasm-function[2749]@[wasm code]
<?>.wasm-function[2744]@[wasm code]
<?>.wasm-function[20461]@[wasm code]
<?>.wasm-function[2739]@[wasm code]
<?>.wasm-function[831]@[wasm code]
<?>.wasm-function[830]@[wasm code]
<?>.wasm-function[847]@[wasm code]
<?>.wasm-function[1977]@[wasm code]
<?>.wasm-function[15600]@[wasm code]
<?>.wasm-function[837]@[wasm code]
<?>.wasm-function[2749]@[wasm code]
<?>.wasm-function[2744]@[wasm code]
<?>.wasm-function[20461]@[wasm code]
<?>.wasm-function[2739]@[wasm code]
<?>.wasm-function[838]@[wasm code]
<?>.wasm-function[2749]@[wasm code]
<?>.wasm-function[2744]@[wasm code]
<?>.wasm-function[20461]@[wasm code]
<?>.wasm-function[835]@[wasm code]
<?>.wasm-function[831]@[wasm code]
<?>.wasm-function[830]@[wasm code]
<?>.wasm-function[847]@[wasm code]
<?>.wasm-function[873]@[wasm code]
<?>.wasm-function[14957]@[wasm code]
<?>.wasm-function[842]@[wasm code]
<?>.wasm-function[354]@[wasm code]
wasm-stub@[wasm code]
__pyproxy_apply@[native code]
http://localhost:8000/pyodide.asm.js:8:2021341
apply@http://localhost:8000/pyodide.asm.js:8:42063
[native code]
pushCode@http://localhost:8000/console.html:15:30
o@https://cdn.jsdelivr.net/npm/jquery.terminal/js/jquery.terminal.min.js:44:91014
T@https://cdn.jsdelivr.net/npm/jquery.terminal/js/jquery.terminal.min.js:44:91831
ENTER@https://cdn.jsdelivr.net/npm/jquery.terminal/js/jquery.terminal.min.js:44:18151
ze@https://cdn.jsdelivr.net/npm/jquery.terminal/js/jquery.terminal.min.js:44:33861
dispatch@https://code.jquery.com/jquery-latest.js:4641:14
handle@https://code.jquery.com/jquery-latest.js:4309:33
```

### CODE REVIEW

**Comment by joseph-long** [comment]
Ah, damn, based it on the wrong branch.

File: packages/astropy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package:
+  name: astropy
+  version: 3.1.2
+
+source:
+  url: https://pypi.io/packages/source/a/astropy/astropy-3.1.2.tar.gz
+  sha256: 4a78a8ec9666d0a51a37f03494aaa5012e241ba37053e6c913c039cddee89ede
+
```
Comment (mdboom) [inline packages/astropy/meta.yaml]: You'll want to put:

```
requirements:
  run:
    - numpy
```

here so that it knows it requires numpy.

Comment (adrn) [reply packages/astropy/meta.yaml:8]: Hey @joseph-long - are you interested in trying out this fix?

Comment (joseph-long) [reply packages/astropy/meta.yaml:8]: Yes, probably still worth a try. I'd need to get my pyodide build env up and running again though... I don't remember now if there was another issue that presented itself after trying @mdboom's suggestion that led me to abandon the attempt.

File: packages/astropy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package:
+  name: astropy
+  version: 3.1.2
+
+source:
+  url: https://pypi.io/packages/source/a/astropy/astropy-3.1.2.tar.gz
```
Comment (rth) [inline packages/astropy/meta.yaml:6]: Would it be worth updating to astropy 4.0 ?

Comment (joseph-long) [reply packages/astropy/meta.yaml:6]: Good point :) I just pushed last night to see if CI would run faster than my laptop build (it did)

**Comment by joseph-long** [comment]
When attempting to build Astropy 4.0.1.post1:

```
Using /src/packages/astropy/build/astropy-4.0.1.post1/.eggs/numpy-1.19.0-py3.8-linux-x86_64.egg
Finished processing dependencies for astropy==4.0.1.post1
emcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/src/cpython/installs/python-3.8.2/include/python3.8 -c astropy/_compiler.c -o build/temp.linux-x86_64-3.8/astropy/_compiler.bc
emcc -O3 -s BINARYEN_METHOD=native-wasm -Werror -s EMULATED_FUNCTION_POINTERS=1 -s EMULATE_FUNCTION_POINTER_CASTS=1 -s SIDE_MODULE=1 -s WASM=1 -s BINARYEN_TRAP_MODE=clamp --memory-init-file 0 -s LINKABLE=1 -s EXPORT_ALL=1 -s ERROR_ON_MISSING_LIBRARIES=0 -pthread -shared build/temp.linux-x86_64-3.8/astropy/_compiler.bc -o build/lib.linux-x86_64-3.8/astropy/compiler_version.cpython-38-x86_64-linux-gnu.wasm
emcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/src/packages/astropy/build/astropy-4.0.1.post1/.eggs/numpy-1.19.0-py3.8-linux-x86_64.egg/numpy/core/include -Icextern/erfa -I/src/cpython/installs/python-3.8.2/include/python3.8 -c astropy/_erfa/ufunc.c -o build/temp.linux-x86_64-3.8/astropy/_erfa/ufunc.bc
In file included from astropy/_erfa/ufunc.c:14:
In file included from /src/packages/astropy/build/astropy-4.0.1.post1/.eggs/numpy-1.19.0-py3.8-linux-x86_64.egg/numpy/core/include/numpy/arrayobject.h:4:
In file included from /src/packages/astropy/build/astropy-4.0.1.post1/.eggs/numpy-1.19.0-py3.8-linux-x86_64.egg/numpy/core/include/numpy/ndarrayobject.h:12:
In file included from /src/packages/astropy/build/astropy-4.0.1.post1/.eggs/numpy-1.19.0-py3.8-linux-x86_64.egg/numpy/core/include/numpy/ndarraytypes.h:6:
/src/packages/astropy/build/astropy-4.0.1.post1/.eggs/numpy-1.19.0-py3.8-linux-x86_64.egg/numpy/core/include/numpy/npy_cpu.h:106:6: error: Unknown CPU, please report this to numpy
      maintainers with information about your platform (OS, CPU and compiler)
    #error Unknown CPU, please report this to numpy maintainers with \
     ^
[... warnings omitted ...]
246 warnings and 1 error generated.
shared:ERROR: compiler frontend failed to generate LLVM bitcode, halting
[2020-07-15 16:46:55] done building package astropy in 669.1 s.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/src/pyodide_build/__main__.py", line 34, in <module>
    main()
  File "/src/pyodide_build/__main__.py", line 28, in main
    args.func(args)
  File "/src/pyodide_build/buildall.py", line 149, in main
    build_packages(packagesdir, outputdir, args)
  File "/src/pyodide_build/buildall.py", line 66, in build_packages
    build_package(pkgname, dependencies, packagesdir, outputdir, args)
  File "/src/pyodide_build/buildall.py", line 21, in build_package
    buildpkg.build_package(packagesdir / pkgname / "meta.yaml", args)
  File "/src/pyodide_build/buildpkg.py", line 249, in build_package
    compile(path, srcpath, pkg, args)
  File "/src/pyodide_build/buildpkg.py", line 141, in compile
    subprocess.run(
  File "/usr/local/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/src/cpython/build/3.8.2/host/bin/python3', '-m', 'pyodide_build', 'pywasmcross', '--cflags', ' ', '--ldflags', '-O3 -s BINARYEN_METHOD=native-wasm -Werror -s EMULATED_FUNCTION_POINTERS=1 -s EMULATE_FUNCTION_POINTER_CASTS=1 -s SIDE_MODULE=1 -s WASM=1 -s BINARYEN_TRAP_MODE=clamp --memory-init-file 0 -s LINKABLE=1 -s EXPORT_ALL=1 -s ERROR_ON_MISSING_LIBRARIES=0 ', '--host', '/src/cpython/build/3.8.2/host', '--target', '/src/cpython/installs/python-3.8.2']' returned non-zero exit status 1.
make[1]: *** [Makefile:5: all] Error 1
make[1]: Leaving directory '/src/packages'
make: *** [Makefile:308: build/packages.json] Error 2
```

It looks like Astropy 4 needs NumPy 1.16 or newer, so the astropy build step is trying to pull in a newer numpy itself and failing because that numpy not patched appropriately.

**Comment by joseph-long** [comment]
I updated the recipe to build 3.2.3, the latest in the 3.x series, since it would be useful to have that incorporated in the short term

**Pull request merged**